### PR TITLE
Remove unused buildcleaner.

### PR DESCRIPTION
**Pull request by drigz**
I can't find any instructions/docs that refer to this. This also lets us remove an unused dependency.
I stumbled across it when looking at the outstanding work for Carsten's
log->slog migration.

### CODE REVIEW

File: third_party/go_repositories.bzl
Comment (faximan) [inline third_party/go_repositories.bzl:1]: Is this change intentional? Maybe add a note about it to the PR description.

**Review by faximan — APPROVED** [review]
Thanks!

File: third_party/go_repositories.bzl
Comment (drigz) [reply third_party/go_repositories.bzl:1]: Yep, it's unused now.
